sql: GA load data logical mode and add experimental physical mode (#13310)
pingcap/docs: sql-statements/sql-statement-load-data.md

The `LOAD DATA` statement batch loads data into a TiDB table.

Starting from TiDB v7.0.0, the `LOAD DATA` SQL statement becomes more powerful by integrating TiDB Lightning's logical import mode, including the following:

- Support importing data from S3 and GCS.
- Support importing Parquet format data.
- Add new parameters `FORMAT`, `FIELDS DEFINED NULL BY`, and `With batch_size=<number>,detached`.

Starting from TiDB v7.1.0, `LOAD DATA` supports the following features:

- Support importing compressed `DELIMITED DATA` and `SQL FILE` data files.

> - Support parrallel import in logical import mode.

Author: ```suggestion
- Support specifying concurrency of the data import in logical import mode.
```

- Support specifying the encoding format of data files through `CharsetOpt`.
- `LOAD DATA` integrates TiDB Lightning's physical import mode. This mode skips the SQL interface, and directly inserts data as key-value pairs into the TiKV nodes, which is an efficient and fast import mode.

> **Warning:**
>

> > The physical import mode in TiDB v7.1.0 is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

Author: ```suggestion
> The concurrency of the data import and physical import mode in TiDB v7.1.0 are experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
```

In physical import mode, `LOAD DATA` writes the locally sorted data to the TiDB [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) subdirectory. The subdirectory naming rule is `import-<tidb-port>/<job-id>`.

Physical import mode currently has not been integrated with [disk resource quota](/tidb-lightning/tidb-lightning-physical-import-mode-usage.md#configure-disk-quota-new-in-v620). Ensure that the corresponding disk has enough space. See [Requirements and restrictions](/tidb-lightning/tidb-lightning-physical-import-mode#requirements-and-restrictions).

### `WITH thread=<number>`

Currently this parameter only applies to logical import mode.

You can specify the concurrency of the data import with `WITH thread=<number>`. The default value is related to `FORMAT`:

> - If `FORMAT` is `PARQUET`, the default value is 75% of the number of CPU cores.
> - For other `FORMAT` options, the default value is the number of logical CPU cores.

Reviewer: ```suggestion
- If `FORMAT` is `PARQUET`, the default value is 75% of the number of CPU cores in the TiDB node.
- For other `FORMAT` options, the default value is the number of cores in the TiDB node.
```

### `WITH batch_size=<number>`

You can specify the number of rows to be written to TiDB in a batch with `WITH batch_size=<number>`. The default value is `1000`. `0` means no splitting.

### `WITH max_write_speed = stringLit`

> When you use physical import mode, you can use this parameter to specify the rate limit for writing to a single TiKV. The default value is `0`, which means no limit.
> 
> This parameter supports the [go-units](https://pkg.go.dev/github.com/docker/go-units#example-RAMInBytes) format. For example, `WITH max_write_speed = '1MB'` means that the maximum write rate to a single TiKV is `1MB/s`.

Reviewer 2: ```suggestion
When using physical import mode, you can use this parameter to specify the rate limit for writing to a single TiKV. The default value is `0`, which means no limit.

This parameter supports the [go-units](https://pkg.go.dev/github.com/docker/go-units#example-RAMInBytes) format. For example, `WITH max_write_speed = '1MB'` specifies a maximum write rate of `1MB/s` to a single TiKV.
```

### `WITH detached`

If you specify an S3 or GCS path, and do not specify the `LOCAL` parameter, you can use `WITH detached` to make `LOAD DATA` run in the background. In this case, `LOAD DATA` returns the task ID.

> You can view the created jobs via [`SHOW LOAD DATA`](/sql-statements/sql-statement-show-load-data.md), or you can use [`CANCEL LOAD DATA` and `DROP LOAD DATA`](/sql-statements/sql-statement-operate-load-data-job.md) to cancel or delete the created jobs.

Reviewer 2: ```suggestion
You can view the created jobs via [`SHOW LOAD DATA`](/sql-statements/sql-statement-show-load-data.md), or you can execute [`CANCEL LOAD DATA` and `DROP LOAD DATA`](/sql-statements/sql-statement-operate-load-data-job.md) to cancel or delete the created jobs.
```
